sendgrid/sendgrid.py

```diff
@@ -58,3 +57,14 @@ def api_key(self, value):
     @property
     def impersonate_subuser(self):
         return self._impersonate_subuser
+
+
+class SendGridClient(SendGridAPIClient):
+    def __init__(self, **opts):
+        super().__init__(**opts)
+
+    def send_mail(self, msg):
+        try:
+            return self.client.mail.send.post(request_body=msg.get())
+        except Exception as e:

```

Comment:
@dibyadas,

What do you think about @w-'s questions about "How are you differentiating say a KeyError or an AttributeError versus an HTTPError?"?